addons/sale_management/models/sale_order.py

```diff
@@ -87,7 +87,7 @@ def update_prices(self):
         self.ensure_one()
         res = super().update_prices()
         for line in self.sale_order_option_ids:
-            line.price_unit = self.pricelist_id.get_product_price(line.product_id, line.quantity, self.partner_id, uom_id=line.uom_id.id)
+            line.write(self._compute_option_data_for_template_change(line))
```

Comment:
After a deeper analysis, `_compute_option_data_for_template_change` was made to receive `sale.order.template.option` records, not `sale.order.option` ones, even if this miraculously works, you should avoid using this method in this case (also, that way, the dict you are giving to write contains unnecessary information which shouldn't be updated, even if it's the same values).

I'd prefer if you follow the logic of `_onchange_product_id` below (you could also make a separate method called by both the onchange and `update_prices`).
```suggestion
        # To compute the discount a so line is created in cache
        values = line._get_values_to_add_to_order()
        new_sol = line.env['sale.order.line'].new(values)
        new_sol._onchange_discount()
        line.discount = new_sol.discount
        if line.order_id.pricelist_id and line.order_id.partner_id:
            line.price_unit = new_sol._get_display_price(product)
```